### DISCUSSION TITLE
Add Right Mouse Button support on Android

### DISCUSSION THREAD

**Discussion by Fl414r** [ppy/osu#22865] [Ideas]
How RMB currently works in Osu!laser on Android (Works like back button)
https://user-images.githubusercontent.com/74704475/224934709-576d46dd-56f3-483e-8271-4127ada0d691.mp4

How RMB should work (using Pojav as an example)
https://user-images.githubusercontent.com/74704475/224935029-66c8ccdc-9f9d-4bd9-a619-33336c0a5a18.mp4

**Comment by ItsShamed** [comment]
Your recording doesn't have touch visualisation, so I can't tell what you are pointing. But support for right-clicking by long tapping is planned. See:
https://github.com/ppy/osu-framework/issues/5200
https://github.com/ppy/osu-framework/pull/5255
https://github.com/ppy/osu/discussions/22869#discussioncomment-5301635

**Reply by Fl414r** [reply]
Because these are not touch, this is a connected external mouse that RMB does not work in Osu. 
(And what do you not see if the cursor works and shows what I am pointing to..?)

**Reply by ItsShamed** [reply]
Sorry, I misunderstood your issue.

---
I tested on my phone and right-clicking via OTG completely works on my end.
Phone model: Xiaomi Redmi Note 8 Pro (begonia)
I simulated OTG input from my PC using https://github.com/Genymobile/scrcpy in OTG mode.

```sh
scrcpy --otg -s <serial>
```

Here is a video proof:

https://user-images.githubusercontent.com/39100084/227626908-855fc539-efac-41fb-a808-a4df45d0de7e.mp4

Although after searching on the internet, right click support depends on your phone model or the hardware you are using.

**Comment by Susko3** [comment]
This is a known problem on some devices, the right mouse button will get mapped to the "back" / Mouse 4 (M4) button. This mapping is done by a manufacturer and the osu! app can't affect it.

If you want to use right click in gameplay, first unmap the `M4` button from the `Back` global action, and then you can map the osu! right button action to `M4` (your right click).

**Reply by Fl414r** [reply]
Do you know that on Android it is impossible to remap the buttons of external devices without Root? It is impossible to get Root on my device, I tried.
I specifically added the Pojav part here to show that it can be fixed, both videos are from Android.

**Reply by Susko3** [reply]
Pojav has a [workaround](https://github.com/PojavLauncherTeam/PojavLauncher/blob/623f7dd3326a653f0e0a75985c03641ab9d5228d/app_pojavlauncher/src/main/java/net/kdt/pojavlaunch/MinecraftGLSurface.java#L473-L476) for your specific device issue. This specific workaround will map the back/Mouse4 button to right click, which is what I have suggested you to do (you can change the mapping in osu! settings).

We don't map this automatically, because it would break devices that work properly and differentiate between the back and right click.
